typings.d.ts

```diff
@@ -110,29 +146,101 @@ declare module 'resin-components' {
 		setRules(rules: FilterRule[]): void;
 	}
 
-	class Filters extends RenderableElementWithProps<FiltersProps, any> {
-		refs: any;
-		render(): JSX.Element | null;
+	class Filters extends RenderableElementWithProps<FiltersProps, any> {}
+
+	class Fixed extends RenderableElementWithProps<DefaultProps, any> {}
+
+	interface FlexProps extends BoxProps {
+		align?: string | string[];
+		justify?: string | string[];
+		direction?: string | string[];
+		wrap?: boolean | boolean[] | string;
+		column?: boolean;
 	}
 
-	interface ProgressBarProps extends DefaultProps {
-		value: number;
-		info?: boolean;
+	class Flex extends RenderableElementWithProps<FlexProps, any> {}
+
+	namespace Heading {
+		class h4 extends RenderableElementWithProps<FlexProps, any> {}
+	}
+
+	interface LinkProps
+		extends DefaultProps,
+			React.AnchorHTMLAttributes<HTMLElement> {
+		blank?: boolean;
+		disabled?: boolean;
 	}
 
-	class ProgressBar extends RenderableElementWithProps<ProgressBarProps, any> {
-		refs: any;
-		render(): JSX.Element | null;
+	class Link extends RenderableElementWithProps<LinkProps, any> {}
+
+	class Input extends RenderableElementWithProps<DefaultProps, any> {}
+
+	class Image extends RenderableElementWithProps<
+		DefaultProps & { src: string },
+		any
+	> {}
+
+	class Modal extends RenderableElementWithProps<
+		DefaultProps & {
+			title: string;
+			action: string;
+			cancel: () => any;
+			done: () => any;
+		},
+		any
+	> {}
+
+	class Navbar extends RenderableElementWithProps<
+		DefaultProps & {
+			brand: JSX.Element;
+		},
+		any
+	> {}
+
+	interface PineTypeModule {
+		rules: {
+			[key: string]: (target: any, value: any) => boolean
+		};
+		validate: (value: any) => boolean;
+		Edit: (props: any) => JSX.Element;
+		Display: (props: any) => JSX.Element;
 	}
 
-	class Provider extends Component<any, any> {
-		refs: any;
-		render(): JSX.Element | null;
+	var PineTypes: { [K in PineDataType]: PineTypeModule }
+
+	class Provider extends Component<any, any> {}
+
+	interface ProgressBarProps extends DefaultProps {
+		value: number;
+		info?: boolean;
 	}
 
+	class ProgressBar extends RenderableElementWithProps<ProgressBarProps, any> {}
+
 	class SchemaSieveClass {
 		filter<T>(items: T[], rule: FilterRule): T[];
 	}
 
 	function SchemaSieve(): SchemaSieveClass;
+
+	class Select extends RenderableElementWithProps<DefaultProps, any> {}
+
+	class Text extends RenderableElementWithProps<DefaultProps, any> {}
+
+	interface TextareaProps extends DefaultProps {
```

Comment:
Would it be possible to also extend `React.TextareaHTMLAttributes` to avoid redefining the following properties? (Even though `value is types as `string | string[] | number` in there.)